plugins/pulp_docker/plugins/registry.py

```diff
@@ -304,9 +305,10 @@ def __init__(self, name, download_config, registry_url, working_dir):
         self.download_config = download_config
         self.registry_url = registry_url
 
-        # Use basic auth information only for retrieving tokens from auth server.
-        self.token_downloader = HTTPThreadedDownloader(self.download_config,
-                                                       AggregatingEventListener())
+        # Use basic auth information for retrieving tokens from auth server and for downloading
```

Comment:
in my understanding you don't need to use basic auth to retrieve the token, you are auth to the server whether with 1) basic auth 2) token auth

Reply:
It does make sense that this isn't necessary for token retrieval, since I have no idea how it worked if the basic auth info was necessary when download_config wasn't being deepcopied.

Since this is the case should I be passing the regular downloader to auth_util.request_token? https://github.com/pulp/pulp_docker/blob/master/plugins/pulp_docker/plugins/registry.py#L481

Reply:
@asmacdo can you share your view here w/r to the auth creds for token retrieval?

Reply:
Basic auth creds should be included in the token request to the auth server.

"The token server should first attempt to authenticate the client using any authentication credentials provided with the request. From Docker 1.11 the Docker engine supports both Basic Authentication and OAuth2 for getting tokens. Docker 1.10 and before, the registry client in the Docker Engine only supports Basic Authentication."
https://docs.docker.com/registry/spec/auth/token/

Reply:
I did a little more digging. I relied on nectar to set the session.auth at __init__ time.
https://github.com/pulp/pulp_docker/pull/160/files#r71227323

But then nectar stopped doing that.
https://github.com/pulp/nectar/commit/4fe7327c4ad1bb2f8c040b75d80306ee047de615

So a deep copy is the right approach. (And should have been from the beginning, sorry!)

Reply:
thanks @asmacdo :star2: :strawberry: